src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -45,31 +45,42 @@ using namespace std;
 SDCPUThread::SDCPUThread(SimpleDataflowCPU* cpu_, ThreadID tid_,
                     System* system_, BaseTLB* itb_, BaseTLB* dtb_,
                     TheISA::ISA* isa_, bool use_kernel_stats_,
-                    bool strict_ser):
+                    unsigned fetch_buf_size, bool strict_ser):
     memIface(*this),
     _cpuPtr(cpu_),
     _name(cpu_->name() + ".thread" + to_string(tid_)),
     strictSerialize(strict_ser),
     _committedState(new SimpleThread(cpu_, tid_, system_, itb_, dtb_, isa_,
                                      use_kernel_stats_)),
-    isa(isa_)
+    isa(isa_),
+    fetchBuf(vector<uint8_t>(fetch_buf_size)),
+    fetchBufMask(~(static_cast<Addr>(fetch_buf_size) - 1))
 {
-    //
+    panic_if(fetch_buf_size % sizeof(TheISA::MachInst) != 0,
+             "Fetch buffer size should be multiple of instruction size!");
+    panic_if(fetch_buf_size > _cpuPtr->cacheLineSize(),
```

Comment:
This is fine for now, but we might want the buffer to be larger than a cache line. If it is larger, then we just send multiple cache-line size requests as the buffer is drained.

Reply:
Hmm, my thought when deciding on the limits was to use this more like the DRAM buffer where we just had the capability to hold the result of one access at a time until we needed something else. I was afraid of going beyond that, because there would come a point where it might make more sense to model any nuances with the instruction cache by attaching a different icache to the instruction port...